skbio/stats/composition.py

```diff
@@ -608,10 +611,269 @@ def centralize(mat):
 
     """
     mat = closure(mat)
-    cen = ss.gmean(mat, axis=0)
+    cen = scipy.stats.gmean(mat, axis=0)
     return perturb_inv(mat, cen)
 
 
+@experimental(as_of="0.4.0-dev")
+def ancom(table, groups,
+          alpha=0.05,
+          tau=0.02,
+          theta=0.1,
+          multiple_comparisons_correction=None,
+          significance_test=scipy.stats.ttest_ind):
+    r""" Performs a differential abundance test using ANCOM
+
+    This is done by calculating pairwise log ratios between all features
+    and performing a signficance test to determine if there is a
+    significant difference in feature ratios with respect to the
+    variable of interest.
+
+    In an experiment with only two treatments, this test tests the
+    following hypothesis for feature :math:`i`
+
+    .. math::
+        H_{0i}:\mathbb{E}[\ln(u_i^{(1)})] = \mathbb{E}[\ln(u_i^{(2)})]
+
+    where :math:`u_i^{(1)}` is the mean abundance for feature
+    :math:`i` in the first group and :math:`u_i^{(2)}` is the
+    mean abundance for feature :math:`i` in the second group.
+
+    This method can be extended to an arbitrary number of classes
+    by passing in a different statistical function.
+
+    Parameters
+    ----------
+    table : pd.DataFrame or array_like
```

Comment:
I think removing that functionality would be best and adding an explicit check that `table` is a `DataFrame` and `grouping` is a `Series` to avoid the behavior noted above.